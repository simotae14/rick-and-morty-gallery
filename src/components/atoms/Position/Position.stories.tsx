import type { Meta, StoryObj } from '@storybook/react';

import Position from './index';

const meta = {
  title: 'Atoms/Position',
  component: Position,
  tags: ['autodocs'],
  argTypes: {
    labelType: {
      control: {
        type: 'text',
      },
    },
    name: {
      control: {
        type: 'text',
      },
    },
    type: {
      control: {
        type: 'text',
      },
    },
    dimension: {
      control: {
        type: 'text',
      },
    },
  },
} satisfies Meta<typeof Position>;

export default meta;

type Story = StoryObj<typeof meta>;

export const Base: Story = {
  args: {
    labelType: 'Origin',
    name: 'Earth (C-137)',
    type: 'Planet',
    dimension: 'Dimension C-137',
  },
};
