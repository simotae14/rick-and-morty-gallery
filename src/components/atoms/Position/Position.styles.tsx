import { styled } from 'styled-components';

export const Info = styled.span`
  display: block;
  font-weight: 800;
`;

export const Value = styled.span`
  font-weight: 400;
`;
