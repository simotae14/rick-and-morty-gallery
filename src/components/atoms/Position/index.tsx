import { Info, Value } from './Position.styles';

interface PositionProps {
  /**
   * Is this the label type of the position
   */
  labelType: string;
  /**
   * Is this the name of the position
   */
  name: string;
  /**
   * Is this the type of the position (planet, space location)
   */
  type: string;
  /**
   * Is this the dimension of the position
   */
  dimension: string;
}

/**
 * Component that contains the information of character locations (Origin, Location)
 */
function Position({ labelType, name, type, dimension }: PositionProps) {
  return (
    <Info>
      {labelType}:{' '}
      <Value>
        {name} - {type} - {dimension}
      </Value>
    </Info>
  );
}

export default Position;
