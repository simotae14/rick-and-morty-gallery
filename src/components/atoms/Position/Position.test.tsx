/* eslint-disable react/jsx-props-no-spreading */
/* eslint-disable import/no-extraneous-dependencies */
import { describe, it } from 'vitest';
// eslint-disable-next-line import/no-extraneous-dependencies
import { render } from '@testing-library/react';
import { axe, toHaveNoViolations } from 'jest-axe';

import Position from '.';

expect.extend(toHaveNoViolations);

const mockPropsLocation = {
  labelType: 'Location',
  name: 'Citadel of Ricks',
  type: 'Space station',
  dimension: 'unknown',
};

describe('Components => atoms => Position', () => {
  it('should render the Position whit the props passed', () => {
    const { asFragment } = render(<Position {...mockPropsLocation} />);
    expect(asFragment()).toMatchSnapshot();
  });

  it('should not fail any accessibility tests the Position with props', async () => {
    const { container } = render(<Position {...mockPropsLocation} />);
    expect(await axe(container)).toHaveNoViolations();
  });
});
