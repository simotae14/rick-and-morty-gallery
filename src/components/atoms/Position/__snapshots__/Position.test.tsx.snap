// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`Components => atoms => Position > should render the Position whit the props passed 1`] = `
<DocumentFragment>
  <span
    class="sc-eLqIir cyjjey"
  >
    Location: 
    <span
      class="sc-HzRMQ crwatE"
    >
      Citadel of Ricks - Space station - unknown
    </span>
  </span>
</DocumentFragment>
`;
